src/dmd/parse.d

```diff
@@ -3497,6 +3498,27 @@ final class Parser(AST) : Lexer
             t = parseVector();
             break;
 
+        case TOK.traits:
+        {
+            AST.TraitsExp te;
+            if (AST.Expression e = parsePrimaryExp())
+                te = cast(AST.TraitsExp) e;
+            if (!te)
+            {
+                // error already emitted while parsing primary
+                t = new AST.TypeError;
+            }
+            else if (strcmp(te.ident.toChars, "getMember") != 0)
+            {
+                // even if this is not a grammar error, it's not worth continuing.
+                error("invalid `__traits`, only `getMember` can be aliased and not `%s`",
+                    te.ident.toChars);
+                t = new AST.TypeError;
+            }
+            else t = new AST.TypeTraits(loc, te);
```

Comment:
Brackets for if/else bodies should be consistent (use `{}` for all or for none).